service/storage_service.cc

```diff
@@ -907,6 +990,11 @@ future<> storage_service::handle_state_normal(inet_address endpoint) {
     // Order Matters, TM.updateHostID() should be called before TM.updateNormalToken(), (see CASSANDRA-4300).
     if (_gossiper.uses_host_id(endpoint)) {
         auto host_id = _gossiper.get_host_id(endpoint);
+        if (is_quarantined(host_id)) {
```

Comment:
Benny, I do not see you have the logic in handle_state_normal to handle who is replacing now.

How will this work?

https://github.com/scylladb/scylladb/pull/12078/files

Reply:
I don't understand what you mean exactly.
Can you be more specific, maybe provide an example of the scenario you're worried about?

Reply:
I mentioned the issue before and asked you to integrate my patch https://github.com/scylladb/scylladb/pull/12078 into your series. 

Anyway, see the example below:

n1, n2, n3
n3 is down
n3 is being replaced by n4
when n4 finishes streaming it advertises n4 as normal status though gossip
when n1 and n2 saw the state normal of n4. Currently n1 and n2 will know n4 replaces n3 because n4 uses the same host id as n3. After this patch, host id of n4 changes, n1 and n2 do not know n4 is replacing by checking host id.

Reply:
OK.